WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthUiState.kt

```diff
@@ -0,0 +1,204 @@
+    open val errorVisibility = false
+    open val contentVisibility = false
+
+    object Scanning : QRCodeAuthUiState() {
+        override val type = SCANNING
+        override val scanningVisibility = true
+    }
+
+    object Loading : QRCodeAuthUiState() {
+        override val type = LOADING
+        override val loadingVisibility = true
+    }
+
+    sealed class Error : QRCodeAuthUiState() {
+        override val type = ERROR
+        override val errorVisibility = true
+        abstract val title: UiString
+        abstract val subtitle: UiString
+        abstract val image: Int
+        open val primaryAction: ErrorPrimaryAction? = null
+        open val secondaryAction: ErrorSecondaryAction? = null
+
+        data class AuthFailed(
+            override val primaryAction: ErrorPrimaryAction,
+            override val secondaryAction: ErrorSecondaryAction
+        ) : Error() {
+            override val type = AUTH_FAILED
+            override val title: UiString = UiStringRes(R.string.qrcode_auth_flow_error_auth_failed_title)
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_error_auth_failed_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_empty_results_216dp
+        }
+
+        data class Expired(
+            override val primaryAction: ErrorPrimaryAction,
+            override val secondaryAction: ErrorSecondaryAction
+        ) : Error() {
+            override val type = EXPIRED
+            override val title: UiString = UiStringRes(R.string.qrcode_auth_flow_error_expired_title)
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_error_expired_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_empty_results_216dp
+        }
+
+        data class InvalidData(
+            override val primaryAction: ErrorPrimaryAction,
+            override val secondaryAction: ErrorSecondaryAction
+        ) : Error() {
+            override val type = INVALID_DATA
+            override val title: UiString = UiStringRes(R.string.qrcode_auth_flow_error_invalid_data_title)
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_error_invalid_data_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_empty_results_216dp
+        }
+
+        data class NoInternet(
+            override val primaryAction: ErrorPrimaryAction,
+            override val secondaryAction: ErrorSecondaryAction
+        ) : Error() {
+            override val type = NO_INTERNET
+            override val title: UiString = UiStringRes(R.string.qrcode_auth_flow_error_no_connection_title)
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_error_no_connection_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_cloud_off_152dp
+        }
+    }
+
+    sealed class Content : QRCodeAuthUiState() {
```

Comment:
Suggestion: We have named the data class in `QRCodeAuthUiState` with a state. `Error` and `Loading`. I was hoping to see `Success` as the data class name here instead of the `Content`.  Let me know if that makes more sense?. Success represents the state better than the `content`, I feel. WDYT?. 🤔

Reply:
@AjeshRPai . Thanks for the comments. I used `Content` instead of `Success` because it encapsulates the content of the screen. I understand the patter 'success' but I'm not 100% it conveys what I want. I'll think on this one and add it to the final PR (flow-part-7) if I decide to change.

Reply:
Sure got it 👍🏼